DeviceConnectionAndReliabilityReadme.md

```diff
@@ -0,0 +1,62 @@
+# Azure IoT Device Client C# SDK
+
+## Device connection and messaging reliability
+
+### Overview
+
+In this document you will find information about
+
+- The connection authentication and renewal methods.
+- The reconnection logic and retry policies.
+- The timeout controls.
+
+### Connection authentication
+
+Authentication can be done using either
+
+- [SAS tokens](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#security-tokens) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/TwinSample/Program.cs#L37)
+- [x509 certificates](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#supported-x509-certificates) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/X509DeviceCertWithChainSample/Program.cs#L43)
+- [Device Provisioning Service](https://docs.microsoft.com/en-us/azure/iot-dps) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/PnpDeviceSamples/Thermostat/Program.cs#L90)
```

Comment:
I think the dps symmetric key sample would work better here: https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/provisioning/Samples/device/SymmetricKeySample/ProvisioningDeviceClientSample.cs

Reply:
I looked at the above sample. It is not the DeviceClient though. It is initializing the Provisioning Client. However, we want to be able to use the DeviceClient using DPS symmetric key right?

Reply:
You do use the symmetric key based security provider to create the client instance: https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/d275140a8116d496e923ee3202ea639dff2a9ee4/provisioning/Samples/device/SymmetricKeySample/ProvisioningDeviceClientSample.cs#L62

Reply:
Sure I can get onboard with that. I think it should matter if it is DeviceClient or some other client but we want to show the use of DeviceAuthenticationWithRegistrySymmetricKey.